dev/com.ibm.ws.security.audit.source/src/com/ibm/ws/security/audit/event/utils/ParameterUtils.java

```diff
@@ -34,19 +34,31 @@ static protected StringBuffer format(Object param) {
                             if (element.getClass().isArray()) {
                                 buf.append(format(element));
                             } else {
-                                buf.append(element.toString());
+                                buf.append(ToString(element));
                             }
                         }
                     }
                     buf.append("]");
                 }
             } else {
-                buf.append(param.toString());
+                buf.append(ToString(param));
             }
         }
         return buf;
     }
 
+    static private String ToString(Object object) {
+	    String value = null;
+	    // it is potential that toString() method throws an exception.
+	    // so if that's the case, catch it and substitution. 
+	    try {
+		    value = object.toString();
+	    } catch (Exception e) {
+		    value = object.getClass().getName();
```

Comment:
Should you add System.identityHashCode() to the class name?  The test case would have to change as well of course if you did that.

Reply:
That's a good idea. I'll modify the code. Thanks.